### PR TITLE
Support Moshi deserialization

### PR DESCRIPTION
**Pull request by babedev**
### What's in this PR?

Introduced Moshi deserialization for Moshi users.

### CODE REVIEW

File: fuel-moshi/src/test/kotlin/com/github/kittinunf/fuel/FuelMoshiTest.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        Fuel.testMode {
+            timeout = 15000
+        }
+    }
+
+    data class HttpBinUserAgentModel(var userAgent: String = "")
+
+    @Test
+    fun gsonTestResponseObject() {
```
Comment (kittinunf) [inline fuel-moshi/src/test/kotlin/com/github/kittinunf/fuel/FuelMoshiTest.kt]: Should we name all test cases here with `moshi`?

Comment (babedev) [reply fuel-moshi/src/test/kotlin/com/github/kittinunf/fuel/FuelMoshiTest.kt]: Oh I forgot.... Will fix immediately.

**Comment by kittinunf** [comment]
💅

**Comment by babedev** [comment]
Should we merge this? 😄

**Comment by kittinunf** [comment]
Yes sure

**Pull request merged**